otter/metrics.py

```diff
@@ -44,54 +50,114 @@
     'deleting FROM scaling_group WHERE "tenantId" IN ({tids})')
 
 
-@defer.inlineCallbacks
+def valid_group_row(row):
+    return (
+        row.get('created_at') is not None and
+        row.get('desired') is not None and
+        row.get('status') not in ('DISABLED', 'ERROR') and
+        not row.get('deleting', False))
+
+
 def get_specific_scaling_groups(client, tenant_ids):
     tids = ', '.join("'{}'".format(tid) for tid in tenant_ids)
     query = QUERY_GROUPS_OF_TENANTS.format(tids=tids)
-    results = yield client.execute(query, {}, ConsistencyLevel.ONE)
-    defer.returnValue(r for r in results
-                      if r.get('created_at') is not None and
-                      r.get('desired') is not None and
-                      r.get('status') not in ('DISABLED', 'ERROR') and
-                      not r.get('deleting', False))
+    d = client.execute(query, {}, ConsistencyLevel.ONE)
+    return d.addCallback(filter(valid_group_row))
+
+
+def get_last_info(fname):
+    eff = Effect(ReadFileLines(fname)).on(
+        lambda lines: (int(lines[0]),
+                       datetime.utcfromtimestamp(float(lines[1]))))
+
+    def log_and_return(e):
+        _eff = err(e, "error reading last tenant")
```

Comment:
"error reading last number of tenants"?  Since It seems to be the length of the tenants, as opposed to the actual tenant ID?